st2common/st2common/runners/paramiko_ssh.py

```diff
@@ -603,15 +606,22 @@ def _get_conninfo_from_ssh_config_for_host(host):
                 user_config_file, e.errno, e.strerror))
 
         user_config = ssh_config.lookup(host)
-        for k in ('hostname', 'username', 'port'):
+        for k in ('hostname', 'port'):
             if k in user_config:
                 ssh_conn_info[k] = user_config[k]
-
+        if 'user' in user_config:
+            ssh_conn_info['username'] = user_config['user']
+        else:
+            raise Exception('User directive missing for host: %s' % ssh_conn_info['hostname'])
```

Comment:
This is not right. There can be an entry without a `user` field. It's a valid SSH entry.